x-pack/plugins/elastic_assistant/server/lib/langchain/elasticsearch_store/elasticsearch_store.ts

```diff
@@ -222,6 +222,13 @@ export class ElasticsearchStore extends VectorStore {
         return getFlattenedHits(maybeEsqlMsearchResponse);
       });
 
+      this.logger.debug(`Similarity search results:\n
+        ${JSON.stringify(
+          results.map((r) => r.metadata.source),
```

Comment:
This interface is correct right? I.e., all docs will have `metadata` and `source`, so no null checking necessary?

Reply:
In case if users add their own KB docs, adding these safety checks:

```results.map((r) => r?.metadata?.source ?? '(missing metadata.source)'),```